### ISSUE TITLE
[OpenApi] List of object is not generated as expected in the interface

### ISSUE THREAD

**Issue by alfredo9f** [sngular/scs-multiapi-plugin#207]
# What happened?

When I define an Api with an endpoint that contains a schema as reference that is an array, the code generated is not correct. For example:

I have a `poductList` endpoint that reference to `Products` component that contains an array of `Product` as response:
![Screenshot from 2023-04-19 15-10-54](https://user-images.githubusercontent.com/126802441/233085280-32a48e8c-3ff6-45a2-a4b2-703572611d0c.png)

I have a `products` endpoint that has an schema that is an array of items where each item is a `Product` component:
![Screenshot from 2023-04-19 15-11-25](https://user-images.githubusercontent.com/126802441/233085286-81bb7c7c-c752-4828-84d8-0937d2c1de08.png)

The components are defined like this:
![Screenshot from 2023-04-19 15-11-41](https://user-images.githubusercontent.com/126802441/233085289-a634be41-e683-4973-93e9-9800c177e8d5.png)

The code generated in the interfaces is:
![Screenshot from 2023-04-19 15-19-19](https://user-images.githubusercontent.com/126802441/233087389-7188fa7c-0e59-4d9a-b274-929adc994c96.png)
![Screenshot from 2023-04-19 15-19-32](https://user-images.githubusercontent.com/126802441/233087396-65028825-4c21-4d0d-bd88-dc3f6b8d1850.png)

# Expected behavior
The return type in both cases should be the same: a list of `ProductDTO`.
In the previous images you can se that when the schema is a component reference in a response, the generated code is not a list of object and only return a simple `ProductDTO`.

# Resources
```yml
openapi: 3.0.3

info:
  title: Customer Position API
  description: A simple Customer position API that allows two operations
    - Get all products
    - Save a product
  version: 1.0-SNAPSHOT

servers:
- url: http://localhost:8080/api/v1/

paths:
  /productsList:
    get:
      summary: get all products
      operationId: getAllProductsList
      tags:
      - products
      responses:
        '200':
          description: A list of products
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Products'
  /products:
    get:
      summary: get all products
      operationId: getAllProducts
      tags:
      - products
      responses:
        '200':
          description: A list of products
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/Product'
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
components:
  schemas:
    Products:
      type: array
      items:
        $ref: '#/components/schemas/Product'
    Product:
      type: object
      required:
      - productId
      - partyId
      - customerId
      - productType
      - externalProductFlag
      properties:
        productId:
          type: number
        partyId:
          type: string
        customerId:
          type: string
        productType:
          type: string
          enum:
          - CURRENT_ACCOUNT
          - SAVING_ACCOUNT
          - CREDIT_CARD
          - MULTI_CURRENCY_ACCOUNT
          - MORTAGE_LOAN
          - CUSTOMER_LOAN
        productSubType:
          type: string
          nullable: true
          enum:
          - SUBTYPE_1
          - SUBTYPE_2
        externalProductFlag:
          type: boolean
    Error:
      type: object
      required:
      - code
      - message
      properties:
        code:
          type: integer
          format: int32
        message:
          type: string
```

**Comment by 5uso-sng** [comment]
Hi! I believe this issue may be the same as #203 . The interface templates need some adjustments in order to support the recently updated embedded types. I've been working on it and it should be fixed soon enough

**Comment by alfredo9f** [comment]
For me this bug is solved. I've tested in version 4.6.4 and works fine!

**Comment by 5uso-sng** [comment]
Fixed in #213
